tests/functional/history/test_show.py

```diff
@@ -56,15 +57,21 @@ def _make_clean_history_recorder(self):
             'awscli.clidriver', history_recorder)
         self._apply_history_recorder_patch(
             'awscli.customizations.history', history_recorder)
+        return history_recorder
 
     def _apply_history_recorder_patch(self, module, history_recorder):
         patch_history_recorder = mock.patch(
             module + '.HISTORY_RECORDER', history_recorder)
         patch_history_recorder.start()
         self.addCleanup(patch_history_recorder.stop)
 
+    def _close_db_writers(self):
```

Comment:
Meh super hesitant to access to internal members? Do we need to do this? Shouldn't everything get cleaned up now?